control-plane/pkg/reconciler/trigger/v2/triggerv2.go

```diff
@@ -85,10 +86,10 @@ func (r *ReconcilerV2) ReconcileKind(ctx context.Context, trigger *eventing.Trig
 
 	consgroup, err := r.reconcileConsumerGroup(ctx, trigger)
 	if err != nil {
-		trigger.Status.MarkDependencyFailed("failed to reconcile consumer group", err.Error())
+		trigger.GetConditionSet().Manage(&trigger.Status).MarkFalse(consumergroup.KafkaConditionConsumerGroupNotReady, "failed to reconcile consumer group", err.Error())
 		return err
 	}
-	trigger.Status.MarkDependencySucceeded()
+	trigger.GetConditionSet().Manage(&trigger.Status).MarkTrue(consumergroup.KafkaConditionConsumerGroupReady)
```

Comment:
The problem with this approach is that the condition `KafkaConditionConsumerGroupNotReady` won't be considered for computing the top-level `Ready` condition by the reconciler framework, can we keep using the `Dependency` condition?